### ISSUE TITLE
Output Normalisation

### ISSUE THREAD

**Issue by pedr0sorio** [pcorrado/DL-Vessel-Localization#2]
Hello,

It is mentioned in the paper that the output structure contains the plane center vectors and normal vectors normalised to be between 0 and 1. I am having difficulty finding in the code the section that executes this. Is this even here?

Thanks in advance.
All the best

**Comment by pcorrado** [comment]
Hi Pedro,

I'm glad to hear you are interested in this work. There is indeed code to restrict the normal vectors and the centers to be between 0 and 1. See OneShotCNN/model.py, line 100, which uses a sigmoid activation function for the last layer of the network, thereby restricting the output to be between 0 and 1. The training data I used (which is not included in the repository), was stored in a csv file with the plane center location normalized by the image size. The normal vectors were stored such that the length of the normal vector was the size of the plane. They were then normalized by the size of the plane (to become normalized unit vectors with direction information only) during training in line 54 of common/utils.py. I hope this helps!

Best,
Phil

**Comment by pedr0sorio** [comment]
Hi Phil,

Thanks for the quick response! I can see it now.

I am quite interested in this work as I am developing a deep learning based CMR planning tool for my master thesis project and your model is built for a quite similar task.

Thank you once more.
All the best,
Pedro

**Comment by pcorrado** [comment]
Sounds very interesting. Good luck with the project!
